tests/dogtag/pytest-ansible/provision/osp_provision.py

```diff
@@ -0,0 +1,206 @@
+        if os.path.isfile(instance_record):
+            with open(instance_record, 'a') as f:
+                f.write('{}\n'.format(server.name))
+        else:
+            with open(instance_record, 'w') as f:
+                f.write('{}\n'.format(server.name))
+
+        # Wait time for VM to come active is 10 min
+        server = conn.compute.wait_for_server(server, status='ACTIVE', failures='ERROR', interval=2, wait=600)
+        return server
+    except Exception as e:
+        print(e)
+        sys.exit(1)
+
+
+def delete_server(conn, server_name):
+    print("Removing Server")
+    server = conn.compute.find_server(server_name)
+    conn.compute.delete_server(server)
+
+
+def list_images(conn, name):
+    print("List Images:")
+    image_list = []
+    for image in conn.image.images():
+        if ("latest" in image.name and image.name.startswith(name)) or (name in image.name):
+            image_list.append(image.name)
+            print("Found Image: {}".format(image.name))
+    return image_list
+
+
+def set_available_network(conn):
+    """
+    This method searches for network with available IPs.
+    And sets the network for server spawining.
+    """
+    print('Getting network availability')
+    global OS_NETWORK_ID
+    for net_id in OS_NETWORK_ID:
+        print net_id
+        used_ips = conn.network.get_network_ip_availability('60cacaff-86a6-4f88-82a4-ed3023724df1').used_ips
+        total_ips = conn.network.get_network_ip_availability('60cacaff-86a6-4f88-82a4-ed3023724df1').total_ips
+        print(used_ips)
+        print(total_ips)
+        if total_ips - used_ips > 10:
+            OS_NETWORK_ID = net_id
+            print('Using network : {} \n Available IPs : {}'.format(net_id, str(total_ips - used_ips)))
+            break
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--image', default='Fedora-Cloud-Base-32', help="Image to provision.")
+@click.option('--inventory', default='/root/host', help='Generate inventory file.')
+@click.option('--image-type', type=click.Choice(['nightly', 'production']), default='nightly',
+              help="Image type (nightly, production)")
+@click.option('--server-type', type=click.Choice(['master', 'client']), default='server',
+              help="Machine type (master, client)")
+@click.option('-v', '--verbose', is_flag=True, help="Verbose")
+def up(inventory, image, image_type, server_type, verbose):
+
+    python_interpreter = ''
+    if not image:
+        image_name = 'Fedora 31'
+    else:
+        image_name = image
+
+    if verbose:
+        openstack.enable_logging(True, stream=sys.stdout)
+
+    conn = create_connection(OS_AUTH_URL, OS_REGION_NAME,
+                             OS_PROJECT_NAME, OS_USERNAME,
+                             OS_PASSWORD)
+    print(image)
+    image_list = list_images(conn, image)
+
+    network = None
+    if image_name in image_list:
```

Comment:
I think the code would be cleaner if written: 

```python3
if image_name not in image_list:
    print("ERROR: No image found")
    sys.exit(1)

print("Using Image: {}".format(image_name))
server = spawn_server(conn, image_name, server_type=server_type)
print(server)

if not server:
    print("ERROR: Unable to provision machine")
    sys.exit(1)

... &c
```

It just reduces indentation and makes error handling clear up front :)

Reply:
Done changes